### ISSUE TITLE
Surveys: Condition questions

### ISSUE THREAD

**Issue by rovats** [khartec/waltz#4705]
Some thoughts on what we may need:

* **Start state**
  * `Enabled/Disabled`
  * `Hidden/Displayed`?
* **Conditions**: List of conditions with the following fields:
  * Question
  * Operator (eq, gt, gte, lt, lte, not null, null etc)
  * Answer value - based on question type
* **Conditions match criteria**
  * `All Match`
  * `Any Match`
* **Target state**: Can be derived automatically from start state

**Comment by davidwatkins73** [comment]
May be worth having a quick discussion on this one.  There is a growing feeling that surveys are not _really_ what is needed.  Instead we should perhaps be thinking about longer-lived documents which have a simpler lifecycle.  A lot of the documents we issue as surveys are really a living document that we wish people to update periodically. I feel surveys are geared more towards fresh data collection.

**Comment by rovats** [comment]
OK, most of the surveys here are fresh data collection exercises.

Also, if we want some surveys to be editable periodically, then based on a flag on the survey run, we could simply allow users to put surveys in edit mode themselves (show another button named `Edit`, which can be used by any survey recipient).

This will allow us to use the survey versioning functionality too.

These user-editable surveys can be displayed as a separate list wherever we display surveys to end users.

Happy to discuss further.

**Comment by davidwatkins73** [comment]
yes, I feel that surveys (the distribution / tracking mechanism) need to be more cleanly separated from surveys (the list of questions and answers).   The surveying (dist/track) mechanism is so similar to the attestations approach that we should probably look to extract them as three distinct parts:

- surveys (q&a's)
- attestations
- campaigns (the dist/tracking)

(Not precious about the naming etc).

**Comment by davidwatkins73** [comment]
Should probably also spin this off as a new issue ;)

**Comment by davidwatkins73** [comment]
![conditional-surveys](https://user-images.githubusercontent.com/576829/88767407-84569d80-d171-11ea-9647-3bdcb6b2b38b.gif)
